compiler/modules/bir/verify.bal

```diff
@@ -151,6 +151,48 @@ function verifyInsn(VerifyContext vc, Insn insn) returns Error? {
     else if insn is ErrorConstructInsn {
         check validOperandString(vc, name, insn.operand, insn.pos);
     }
+    else if insn is TypeMergeInsn {
+        check verifyTypeMerge(vc, insn);
+    }
+    else if insn is TypeBranchInsn {
+        check verifyTypeBranch(vc, insn);
+    }
+}
+
+function verifyTypeMerge(VerifyContext vc, TypeMergeInsn insn) returns err:Internal? {
+    Register unnarrowedOp = unnarrow(insn.result);
+    SemType union = t:NEVER;
+    foreach Register r in insn.operands {
+        if unnarrowedOp != unnarrow(r) {
```

Comment:
This comparison checks the whole struct. It will be faster to `===` or just check the number. Let's just check if the number is the same. `unnarrowedOp.number != unnarrow(r)number`